src/NuGet.Services.AzureSearch/SearchService/SearchResponseBuilder.cs

```diff
@@ -110,6 +111,50 @@ public class SearchResponseBuilder : ISearchResponseBuilder
             };
         }
 
+        public AutocompleteResponse AutocompleteFromSearch(
+            AutocompleteRequest request,
+            SearchParameters parameters,
+            string text,
+            DocumentSearchResult<SearchDocument.Full> result,
+            TimeSpan duration)
+        {
+            var results = result.Results;
+            result.Results = null;
+
+            List<string> data;
+            switch (request.Type)
+            {
+                case AutocompleteRequestType.PackageIds:
+                    data = results.Select(x => x.Document.PackageId).ToList();
+                    break;
+
+                case AutocompleteRequestType.PackageVersions:
+                    data = results.SelectMany(x => x.Document.Versions).ToList();
+                    break;
+
+                default:
+                    throw new InvalidOperationException($"Unknown autocomplete request type '{request.Type}'");
+            }
+
+            return new AutocompleteResponse
+            {
+                Context = new AutocompleteContext
+                {
+                    Vocab = "http://schema.nuget.org/schema#",
+                },
+                TotalHits = result.Count.Value,
+                Data = data,
+                Debug = DebugInformation.CreateOrNull(
+                    request,
+                    _options.Value.SearchIndexName,
+                    parameters,
+                    text,
+                    result,
+                    duration,
+                    AuxiliaryData.Metadata),
```

Comment:
Hm... sending back aux metadata is a little weird in this case since it is unused in this code path. Maybe set it to null here.

Reply:
Fair enough. Removed